react-components/select/step6.md

```diff
@@ -0,0 +1,53 @@
+Typeahead variants support adding new options, based on the user entered input. This is supported in two parts, first by allowing a user to select their custom input, and second by allowing this custom input to be added to the overall list of options for permanence.
+
+1) **Add the isCreateable flag to the Select component.**
+
+<pre class="file" data-target="clipboard">
+  isCreatable
+</pre>
+
+This is what your select component should look like:
+```
+<Select 
+  onToggle={this.onToggle}
+  onSelect={this.onSelect}
+  selections={selected}
+  isExpanded={isExpanded}
+  placeholderText="Select a state"
+  variant={SelectVariant.typeahead}
+  onClear={this.clearSelection}
+  onFilter={this.customFilter}
+  isCreatable
+>
+    {options}
+</Select>
+```
+
+<strong>Note:</strong> With this flag, a new message is displayed at the end of the list, prompting the user to create an option based on the input. Additionally, this message replaces the "no results found" message when filtering would otherwise return nothing. This lets you use your current value as a new selection, but does not persist the data - it will appear as selected, but will not appear in the overall list.
+
+<strong>Hint:</strong> To persist a new value permanently, use the onCreateOption property.
```

Comment:
How should this hint be applied? Should it be applied in this step or another time in the future?

Reply:
It's more transitional text than a hint, thinking about it. The next step is about persisting the new options. I'm going to remove it.